backend/api/src/id.rs

```diff
@@ -76,17 +89,22 @@ where
 
     fn from_input_value(value: &juniper::InputValue) -> Option<Self> {
         let s = value.as_string_value()?;
-        if s.len() != 12 {
-            return None;
-        }
 
-        let s = s.as_bytes();
-        let kind = [s[0], s[1]];
-        let hi = decode_base85(s[2..7].try_into().unwrap())?;
-        let lo = decode_base85(s[7..12].try_into().unwrap())?;
-        let key = ((hi as u64) << 32) | lo as u64;
+        let out = (|| -> Option<Self> {
```

Comment:
Is the explicit return type really necessary?

Anyway, this looks a bit weird to my admittedly under developed Rust-aesthetic senses. Why not just early-return `Some(Self::invalid())` outside of the closure as before?

I'm also thinking that this whole closure should probably go in a "constructor", anyway, right? Something like

```rust
impl Id {
    fn from_string(string: &str) -> Self { ... }
}
```

Reply:
> Is the explicit return type really necessary?

Yes. But good point. I implemented `FromStr` now and use that in the function. Looks nicer.